test-result-summary-client/src/Build/TopLevelBuildTable.jsx

```diff
@@ -56,16 +53,21 @@ export default class TopLevelBuildTable extends Component {
 
             await Promise.all(buildInfo.slice(i, pageSize * currentPage).map(async build => {
                 if (build.totals) return;
-                const fetchBuild = await fetch(`/api/getTotals?buildName=${buildName}&url=${url}&buildNum=${build.build.buildNum}`, {
-                    method: 'get'
-                });
-                const totals = await fetchBuild.json();
+                const totals = await this.fetchBuilds(`/api/getTotals?buildName=${buildName}&url=${url}&buildNum=${build.build.buildNum}`);
                 build.totals = totals;
             }));
             this.forceUpdate();
         }
     }
 
+    async fetchBuilds(buildUrl) {
+        const fetchBuild = await fetch(buildUrl, {
+            method: 'get'
+        });
+        const builds = await fetchBuild.json();
+        return builds;
+    }
```

Comment:
Thanks for working on this, @avishreekh ! Please see the comments below:
- As mentioned in https://github.com/adoptium/aqa-test-tools/issues/289#issuecomment-812052457, this should be in the common utils file. It will be used for all fetch calls.
- It depends on the API call, we may fetch other data.
`async fetchBuilds(buildUrl) {` -> `async fetchData(url) {`
- The purpose of having a common function is to add error handling. Something as following:
```
try {
  const response = await fetch(url, {
     method: 'get'
  });
  const jsonResponse = await response.json();
  return jsonResponse;
} catch (e) {
   console.error(e);
}
```

Reply:
Thank you for your valuable comments @llxia. I will incorporate these changes and let you know.

Reply:
I have attempted to incorporate all the requested changes. Please let me know if anything else is required @llxia